metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -188,70 +181,65 @@ final class BloopServers(
             reconnect().ignoreValue
         }
 
-      case item
+      case (item, Some(bloopPath))
           if item == Messages.BloopGlobalJsonFilePremodified.openGlobalJsonFile =>
         val position = new Position(0, 0)
         val range = new org.eclipse.lsp4j.Range(position, position)
         val command = ClientCommands.GotoLocation.toExecuteCommandParams(
           ClientCommands.WindowLocation(
-            bloopGlobalJsonFilePath.toURI.toString,
+            bloopPath.toURI.toString,
             range
           )
         )
         Future.successful(languageClient.metalsExecuteClientCommand(command))
 
-      case item
-          if item == Messages.BloopGlobalJsonFilePremodified.useGlobalFile =>
-        Future.unit
+      case _ => Future.unit
+
     }
   }
 
   private def updateBloopGlobalJsonFileThenRestart(
-      bloopGlobalJsonFilePath: AbsolutePath,
-      bloopCreatedByMetalsFilePath: AbsolutePath,
-      maybeBloopJvmProperties: Option[List[String]],
+      maybeBloopJvmProperties: List[String],
       maybeJavaHome: Option[String],
-      bloopJsonUpdateCause: BloopJsonUpdateCause,
-      reconnect: () => Future[BuildChange]
+      reconnect: () => Future[BuildChange],
+      bloopJsonUpdateCause: BloopJsonUpdateCause
   ): Future[Unit] = {
-    writeJVMPropertiesToBloopGlobalJsonFile(
-      bloopGlobalJsonFilePath,
-      bloopCreatedByMetalsFilePath,
-      maybeBloopJvmProperties,
-      maybeJavaHome
-    ) match {
-      case Failure(exception) => Future.failed(exception)
-      case Success(_) =>
-        languageClient
-          .showMessageRequest(
-            Messages.BloopJvmPropertiesChange.params(bloopJsonUpdateCause)
-          )
-          .asScala
-          .flatMap {
-            case messageActionItem
-                if messageActionItem == Messages.BloopJvmPropertiesChange.reconnect =>
+    languageClient
+      .showMessageRequest(
+        Messages.BloopJvmPropertiesChange.params(bloopJsonUpdateCause)
+      )
+      .asScala
+      .flatMap {
+        case messageActionItem
+            if messageActionItem == Messages.BloopJvmPropertiesChange.reconnect =>
+          writeJVMPropertiesToBloopGlobalJsonFile(
```

Comment:
Changed the order here to first ask then update to avoid race conditions which might be what @zmerr was seeing